File: 2021/Day01/Day01.c
#include "../input.h"
#include <xlocale.h>
#define WINDOWSZIE 3

int getWindow(char **input, int index) {
    int sum = 0;
    for(int i=index; i<index+WINDOWSZIE; i++)
        sum += atoi(input[i]);
    return sum;
}

int main(int argc, char **argv) {
    // ---------- //
    // read input //
    // ---------- //
    int len, count = 0;
    char **input = readInput("data/input01.txt", &len);

    // -------- //
    // part one //
    // -------- //
    for(int i=1; i<len; i++)
        if (atoi(input[i-1]) < atoi(input[i])) 
            count++;
    printf("Part 1: %d\n", count);

    // -------- //
    // part two //
    // -------- //
    count = 0;
    for(int i=1; i<len-WINDOWSZIE+1; i++)
        if (getWindow(input, i - 1) < getWindow(input, i))
            count++;
    printf("Part 2: %d\n", count);
}

File: 2024/Day01.py
with open('2024/data/input01.txt') as f: 
    [left, right] = [sorted(l) for l in zip(*[list(map(int, line.split("   "))) for line in f])]

print("Part 1:", sum([abs(x-y) for x,y in zip(left, right)]))
print("Part 2:", sum(l * right.count(l) for l in set(left)))
